### PR TITLE
Feature/refine api for dygraph

### PR DESCRIPTION
**Pull request by JiabinYang**
This PR add api doc for digraph:

1. `BackwardStrategy`
2. `no_grad`
3. `guard`
4. `to_variable`
5. `Embedding`
6. `NCE`
7. `Prelu`

### CODE REVIEW

File: python/paddle/fluid/dygraph/base.py [hunk 1]
```diff
@@ -44,6 +44,32 @@ def _switch_tracer_mode_guard_(is_train=True):
...
+                inp = np.ones([3, 32, 32], dtype='float32')
+                t = fluid.dygraph.base.to_variable(inp)
+                fc1 = fluid.FC('fc1', size=4, bias_attr=False, num_flatten_dims=1)
+                fc2 = fluid.FC('fc2', size=4)
+                ret = fc1(t)
+                dy_ret = fc2(ret)
+
+        test_layer()
+
+    :param func
```
Comment (junjun315) [inline python/paddle/fluid/dygraph/base.py]: what?

File: python/paddle/fluid/dygraph/base.py [hunk 2]
```diff
@@ -74,6 +125,29 @@ def guard(place=None):
...
+    """
+    This function will create a variable from ndarray
+
+    Args:
+        value(ndarray): the numpy value need to be convert
+        block(fluid.Block|None): which block this variable will be in
+        name(str|None): Name of Varaible
+
+    return:
+        None
```
Comment (junjun315) [inline python/paddle/fluid/dygraph/base.py]: return var?

File: python/paddle/fluid/dygraph/nn.py [hunk 3]
```diff
@@ -1189,12 +1182,18 @@ class Embedding(layers.Layer):
...
 
     Examples:
+
         .. code-block:: python
 
-          dict_size = len(dataset.ids)
-          input = fluid.layers.data(name='ids', shape=[32, 32], dtype='float32')
-          embedding = fluid.Embedding(size=[dict_size, 16])
-          fc = embedding(input)
+          inp_word = np.array([[[1]]]).astype('int64')
```
Comment (junjun315) [inline python/paddle/fluid/dygraph/nn.py:1188]: `[[[1]]]` or `[[1]]` ?

File: python/paddle/fluid/dygraph/base.py (hunk 1 above)
Comment (JiabinYang) [reply python/paddle/fluid/dygraph/base.py]: this one is a decorator so it’s parameter should be func

File: python/paddle/fluid/dygraph/nn.py (hunk 3 above)
Comment (JiabinYang) [reply python/paddle/fluid/dygraph/nn.py:1188]: I tested this one

**Pull request merged**